src/Streamly/Internal/LZ4.hs

```diff
@@ -210,24 +213,30 @@ compressChunk Config{..} speed ctx arr = do
               -- data and the next 4 bytes store the length of the
               -- compressed data.
               (MArray.Array fptr dstBegin dstMax) <-
-                  MArray.newArray (maxCompLen + metaSize)
-              let hdrCompLen = dstBegin `plusPtr` compSizeOffset
-                  compData = dstBegin `plusPtr` dataOffset
+                  MArray.newArray (maxCompLen + metaSize_)
+              let hdrCompLen = dstBegin `plusPtr` compSizeOffset_
+                  compData = dstBegin `plusPtr` dataOffset_
               compLenC <-
                   c_compressFastContinue
                       ctx src compData uncompLenC maxCompLenC speedC
               when (compLenC <= 0)
                 $ error $ "compressChunk: c_compressFastContinue failed. "
                     ++ "uncompLenC: " ++ show uncompLenC
                     ++ "compLenC: " ++ show compLenC
-              setUncompSize dstBegin (cIntToI32 uncompLenC)
+              setUncompSize_ dstBegin (cIntToI32 uncompLenC)
               poke hdrCompLen (cIntToI32 compLenC)
               let compLen = cIntToInt compLenC
-                  dstEnd = dstBegin `plusPtr` (compLen + metaSize)
+                  dstEnd = dstBegin `plusPtr` (compLen + metaSize_)
                   compArr = MArray.Array fptr dstEnd dstMax
               -- It is safe to shrink here as we need to hold the last 64KB of
               -- the previous uncompressed array and not the compressed one.
               Array.unsafeFreeze <$> MArray.shrinkToFit compArr
+    where
+
+    metaSize_ = metaSize cfg
```

Comment:
Why not use NamedFieldPuns instead?

Reply:
They aren't record fields anymore. They are just plain functions.